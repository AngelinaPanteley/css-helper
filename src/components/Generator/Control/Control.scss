@import '../../../variables';

.NumberControl {
  width: 90%;
  margin: 20px auto;
}

.Label {
  @include flex-wrapper;
  
  color: $color-white;
  margin-bottom: 10px;
}