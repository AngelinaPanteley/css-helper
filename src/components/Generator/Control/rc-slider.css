.rc-slider-handle:active {
  border-color: #ffffff;
  box-shadow: 0 0 5px #ffffff;
}

.rc-slider-track {
  background-color: #842a50;
  height: 7px;
}

.rc-slider-rail {
  background-color: #c41d55;
  height: 7px;
}

.rc-slider-step {
  height: 7px;
}

.rc-slider-handle {
  border: solid 2px #ffffff;
  background-color: #ffffff;
  width: 17px;
  height: 17px;
}

.rc-slider-handle-click-focused,
.rc-slider-handle-click-focused:focus {
  box-shadow: 0 0 1px 1px #fff;
}

.rc-slider-handle:hover {
  box-shadow: 0 0 1px 1px #fff;
  border-color: #fff;
}

.rc-slider-handle:focus {
  border-color: #fff;
  box-shadow: 0 0 1px 1px #fff;
}

.rc-slider-tooltip-inner {
  background-color: #c41d55;
}

.rc-slider-tooltip-placement-top .rc-slider-tooltip-arrow {
  border-top-color: #c31c55;
}

.rc-slider-output {
  position: absolute;
  bottom: 23px;
  right: 0;
  color: #fff;
}

.chrome-picker {
  margin: auto;
}