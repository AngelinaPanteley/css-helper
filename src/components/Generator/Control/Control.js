import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Control.scss';
import 'rc-slider/assets/index.css';
import 'rc-tooltip/assets/bootstrap.css';
import './rc-slider.css';
import * as controlTypes from '../controls';
import { SliderPicker } from 'react-color';
import Slider from 'rc-slider';

const createSliderWithTooltip = Slider.createSliderWithTooltip;
const SliderWithTooltip = createSliderWithTooltip(Slider);

class Control extends PureComponent {
  static propsTypes = {
    control: PropTypes.shape({
      type: PropTypes.string.isRequired,
      initialValue: PropTypes.oneOfType([PropTypes.string,
      PropTypes.number]).isRequired
    }).isRequired,
    handleChange: PropTypes.func.isRequired,
    value: PropTypes.oneOfType([PropTypes.string,
    PropTypes.number]).isRequired,
  }

  render() {
    const control = this.props.control;
    let renderControl = null;

    switch (control.type) {
      case controlTypes.number:
        renderControl = <SliderWithTooltip
          min={control.minValue}
          max={control.maxValue}
          defaultValue={control.initialValue}
          value={this.props.value}
          tipFormatter={value => `${value}${control.units}`}
          onChange={this.props.handleChange} />;
        break;
      case controlTypes.color:
        renderControl = <SliderPicker
          color={this.props.value}
          onChangeComplete={(value) => {
            this.props.handleChange(value.hex)
          }} />;
        break;
      default: break;
    }
    return (
      <div className={styles.NumberControl} >
        <label className={styles.Label}>
          <span>{control.label}</span>
          <span>{this.props.value}{control.units}</span>
        </label>
        {renderControl}
      </div>
    );
  }
}

export default Control;