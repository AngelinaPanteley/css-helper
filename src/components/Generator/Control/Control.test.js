"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Control from './Control.js';
import reducer from '../../../store/reducers/index';

configure({ adapter: new Adapter() });

const store = createStore(reducer);

describe('>>>Control', () => {

  //SNAPSHOTS

  const props = {
    control: {
      type: 'number',
      initialValue: 30,
    },
    handleChange: () => { },
    value: 30,
  };

  const controlComponent = <Provider store={store}><MemoryRouter>
    <Control {...props} />
  </MemoryRouter></Provider>

  const component = renderer.create(controlComponent);

  it('renders control correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
