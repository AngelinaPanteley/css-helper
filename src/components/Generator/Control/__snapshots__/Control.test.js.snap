// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control renders control correctly 1`] = `
<div>
  <label>
    <span />
    <span>
      30
    </span>
  </label>
  <div
    className="rc-slider"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="rc-slider-rail"
      style={Object {}}
    />
    <div
      className="rc-slider-track"
      style={
        Object {
          "left": "0%",
          "width": "30%",
        }
      }
    />
    <div
      className="rc-slider-step"
    />
    <div
      aria-disabled={false}
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={30}
      className="rc-slider-handle"
      onBlur={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="slider"
      style={
        Object {
          "left": "30%",
        }
      }
      tabIndex={0}
    />
    <div
      className="rc-slider-mark"
    />
  </div>
</div>
`;
